includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,481 @@
+		add_action( 'woocommerce_order_status_changed', __CLASS__ . '::order_status_changed', 10, 4 );
+
+		// Add meta boxes
+		add_action( 'add_meta_boxes', __CLASS__ . '::add_meta_boxes' );
+
+		// Add action buttons
+		add_action( 'woocommerce_order_item_add_action_buttons', __CLASS__ . '::add_action_buttons', 10, 1 );
+
+		// Add scripts and styles for admin
+		add_action( 'admin_enqueue_scripts', __CLASS__ . '::admin_enqueue_scripts' );
+
+		// Add Admin Backend Actions
+		add_action( 'wp_ajax_swedbank_pay_capture', [ $this, 'ajax_swedbank_pay_capture' ] );
+
+		add_action( 'wp_ajax_swedbank_pay_cancel', [ $this, 'ajax_swedbank_pay_cancel' ] );
+
+		// Filters
+		add_filter( 'swedbank_pay_generate_uuid', [ $this, 'generate_uuid' ], 10, 1 );
+		add_filter( 'swedbank_pay_payment_description', [ $this, 'payment_description' ], 10, 2 );
+
+		// Process swedbank queue
+		if ( ! is_multisite() ) {
+			add_action( 'customize_save_after', [ $this, 'maybe_process_queue' ] );
+			add_action( 'after_switch_theme', [ $this, 'maybe_process_queue' ] );
+		}
+
+		// Add admin menu
+		add_action( 'admin_menu', [ $this, 'admin_menu' ], 99 );
+
+		// Add Upgrade Notice
+		if ( version_compare( get_option( 'woocommerce_payex_psp_version', '1.2.0' ), '1.2.0', '<' ) ) {
+			add_action( 'admin_notices', __CLASS__ . '::upgrade_notice' );
+		}
+	}
+
+	public function includes() {
+		$vendorsDir = dirname( __FILE__ ) . '/../vendors';
+
+		if ( ! class_exists( '\\SwedbankPay\\Api\\Client\\Client', false ) ) {
+			require_once $vendorsDir.'/swedbank-pay-sdk-php/vendor/autoload.php';
+		}
+
+		if ( ! class_exists( '\\Ramsey\\Uuid\\Uuid', false ) ) {
+			require_once $vendorsDir . '/ramsey-uuid/vendor/autoload.php';
+		}
+
+		if ( ! class_exists( 'FullNameParser', false ) ) {
+			require_once $vendorsDir . '/php-name-parser/vendor/autoload.php';
+		}
+
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-reference.php' );
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-transactions.php' );
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-queue.php' );
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-icon.php' );
+	}
+
+	/**
+	 * Install
+	 */
+	public function install() {
+		// Install Schema
+		WC_Swedbank_Pay_Transactions::instance()->install_schema();
+
+		// Set Version
+		if ( ! get_option( 'woocommerce_payex_psp_version' ) ) {
+			add_option( 'woocommerce_payex_psp_version', '1.1.0' );
+		}
+	}
+
+	/**
+	 * Init localisations and files
+	 */
+	public function init() {
+		// Functions
+		include_once( dirname( __FILE__ ) . '/functions-swedbank-pay.php' );
+	}
+
+	/**
+	 * WooCommerce Init
+	 */
+	public function woocommerce_init() {
+		include_once( dirname( __FILE__ ) . '/class-wc-background-swedbank-pay-queue.php' );
+		self::$background_process = new WC_Background_Swedbank_Pay_Queue();
+	}
+
+	/**
+	 * WooCommerce Loaded: load classes
+	 */
+	public function woocommerce_hook_loaded() {
+		// Includes
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ["include_once" statement detected. File manipulations are discouraged. Statement is not a function, no parentheses are required. Concatenating is forbidden.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)